src/clickhouse/stores.ts

```diff
@@ -0,0 +1,29 @@
+import client from "./createClient.js";
+
+class ClickhouseStore {
+  private ChainsPromise: Promise<string[]> | undefined = undefined;
+
+  constructor() {
+    // Fetch data initially
+    this.fetchData();
+
+    // Set up a timer to fetch data periodically (e.g., every 1 hour)
+    setInterval(() => {
+      this.fetchData();
+    }, 10000); // 3600000 milliseconds = 1 hour
+  }
+
+  private fetchData() {
+    this.ChainsPromise = client
+      .query({ query: "SELECT DISTINCT chain FROM blocks", format: "JSONEachRow" })
```

Comment:
I think we could replace `blocks` => `module_hashes` and it will be much faster since there's only a few rows